pkg/services/secrets/kvstore/kvstore.go

```diff
@@ -51,19 +50,20 @@ func ProvideService(
 			}
 			logger.Error("error starting secrets plugin, falling back to SQL implementation")
 		} else {
-			store = &secretsKVStorePlugin{
+			pluginStore := NewCachedKVStore(&secretsKVStorePlugin{
 				secretsPlugin:                  secretsPlugin,
 				secretsService:                 secretsService,
 				log:                            logger,
 				kvstore:                        namespacedKVStore,
 				backwardsCompatibilityDisabled: features.IsEnabled(featuremgmt.FlagDisableSecretsCompatibility),
-			}
+			}, 5*time.Second, 5*time.Minute)
+			return NewKVStoreWithFallback(pluginStore, defaultStore), nil
```

Comment:
Every time we have a Secret plugin configured we would need a Fallback? What if migration will not run?

Reply:
Good point, maybe the `useFallback` flag in the wrapper could be set based on the migration flag

Reply:
In that case the fallback could be disabled by default, but I think it's easier to return the wrapper than have a bunch of if and elses and return a different version of secret store each time.

In the future once we have LockAndRelease and migration as a background service Migration code should run every time, and the check for if is needed is done inside the Migrate code, which could also set or unset the feedback usage depending on those coditions.